Sources/Fuzzilli/Core/ProgramBuilder.swift

```diff
@@ -757,6 +756,14 @@ public class ProgramBuilder {
             }
         }
 
+        // Ensure that all required instructions are being adopted into a valid scope
+        for idx in requiredInstructions {
```

Comment:
That's true but you can also end up in a situation where the instruction being spliced has an input that is the result of an operation that has a non-default context. For example:
```
BeginAsyncFunction
...
v1 = Await v0
CallFunction(x, v1)
...
EndAsyncFunction
```
If we were to splice `CallFunction` the input re-wiring logic would also include the `Await` instruction as that is one of the inputs to `CallFunction`.

There were two approaches that I considered:
1. Drop the splice and return
2. Replace instructions that require a non-default context with LoadNull or LoadUndefined instructions.

I think the former seemed simpler and straightforward to implement. The latter whilst doable I wasn't fully convinced that just null-ing out inputs would be any better than just dropping the splice.

Reply:
Right, ok, correct. That's unfortunate. 

So I think the proper way to handle this would be to include surrounding Blocks in the slice if they are required to create the necessary context. The logic for that is probably not exactly trivial though :( It'd probably involve keeping track of context requirements in the loop that computes `requiredInstructions`, and including the next block that creates a necessary context if that context is not currently active.

I think for now we can also go with your approach and make a comment about how this should probably work in the future though.